addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -624,7 +624,7 @@ export const editorCommands = {
         const fontsSet = new Set(fonts);
         for (const font of fontsSet) {
             colorElement(font, color, mode);
-            if (!hasColor(font, mode) && !font.hasAttribute('style')) {
+            if ((!hasColor(font, 'color') && !hasColor(font,'backgroundColor')) && (!font.hasAttribute('style') || !color)) {
```

Comment:
@maad-odoo `(!font.hasAttribute('style')` i believe this condition is not needed here when this code executes for color action, instead check for the class present that's applying the color to the text in the font tag in our case `class='bg-*'`. This condition needs to be optimized, look into it.